lib/inputstreamhelper/kodiutils.py

```diff
@@ -16,6 +16,16 @@
 class progress_dialog(DialogProgress, object):  # pylint: disable=invalid-name,useless-object-inheritance
     """Show Kodi's Progress dialog"""
 
+    def __init__(self):
+        """Initialize a new progress dialog"""
+        # Wait for previous Progress dialog to close
+        # Progress dialog Window ID is 10101: https://kodi.wiki/view/Window_IDs
+        previous_active = get_current_window_id() == 10101
+        while previous_active:
```

Comment:
All in all a good solution, just one thing about the style:
Why don't you put `while get_current_window_id() == 10101:` here and remove both lines defining `previous_active`?

Reply:
Thanks for noticing, that's much better.